common/src/main/java/io/github/gaming32/bingo/util/BingoUtil.java

```diff
@@ -184,4 +191,44 @@ public static Component getDisplayName(PlayerTeam team, PlayerList playerList) {
         }
         return team.getDisplayName();
     }
+
+    public static boolean isDyeableArmor(Item item) {
+        return item instanceof ArmorItem armor && armor.getMaterial()
+            .value()
+            .layers()
+            .stream()
+            .anyMatch(ArmorMaterial.Layer::dyeable);
+    }
+
+    public static ItemStack setPotion(ItemStack stack, Holder<Potion> potion) {
+        stack.set(DataComponents.POTION_CONTENTS, new PotionContents(potion));
+        return stack;
+    }
+
+    public static boolean collidesWithProjectedBox(Vec3 entityOrigin, Vec3 boxNormal, double boxWidth) {
+        // Distance from origin to the closest point on the boxNormal line
+        final double pointDistance = entityOrigin.dot(boxNormal) / boxNormal.lengthSqr();
+
+        // Entity is behind box
+        if (pointDistance < 0) {
+            return false;
+        }
+
+        final Vec3 closestPoint = boxNormal.scale(pointDistance);
+        final double angle = vectorAngle(closestPoint, entityOrigin);
+        final double maxDistance = distanceToSquareEdge(angle, boxWidth / 2);
+        return closestPoint.distanceToSqr(entityOrigin) <= maxDistance * maxDistance;
+    }
+
+    public static double vectorAngle(Vec3 a, Vec3 b) {
+        return Math.atan2(a.dot(b), a.length() * b.length());
```

Comment:
Not sure where you got this from, but I'm pretty sure it's wrong if you want to get the angle between two vectors.
```suggestion
        return Math.acos(a.dot(b) / (a.length() * b.length()));
```
is the correct formula for this (also add in a check for if the vector length is nearly 0). Also good to document that this returns radians rather than degrees.